modules/unsupported/geojsonstore/src/main/java/org/geotools/data/geojson/GeoJSONFeatureSource.java

```diff
@@ -36,7 +36,8 @@
 public class GeoJSONFeatureSource extends ContentFeatureSource {
     private static final Logger LOGGER = Logging.getLogger(GeoJSONFeatureSource.class);
 
-    private FeatureCollection<?, ?> collection = null;
+    private FeatureCollection<SimpleFeatureType, SimpleFeature> collection;
```

Comment:
This caching could cause OOM in the long run and/or cause the data to get out of synch with the file system. Maybe add  a store variable to control it? Just thinking out loud here, not asking for a change.

Reply:
the caching **should** only occur if you ask for all the features directly - access via a FeatureIterator will stream the features.